packages/fx-core/src/component/generator/apiPluginFromScratch/generator.ts

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * @author yuqzho@microsoft.com
+ */
+
+import { Context, FxError, Inputs, ok, Result } from "@microsoft/teamsfx-api";
+import { DefaultTemplateGenerator } from "../templates/templateGenerator";
+import {
+  ApiAuthOptions,
+  ApiPluginStartOptions,
+  CapabilityOptions,
+  ProgrammingLanguage,
+  QuestionNames,
+} from "../../../question";
+import { ActionContext } from "../../middleware/actionExecutionMW";
+import { Generator } from "../generator";
+import { merge } from "lodash";
+import { TemplateNames } from "../templates/templateNames";
+import { TemplateInfo } from "../templates/templateInfo";
+
+const enum telemetryProperties {
+  templateName = "template-name",
+  isDeclarativeCopilot = "is-declarative-copilot",
+}
+
+export class ApiPluginFromScratchGenerator extends DefaultTemplateGenerator {
```

Comment:
A small suggestion: can we call it `CopilotExtensionGenerator`?

Reply:
I want to differentiate it from the SpecGenerator in apispec folder. so maybe copilotExtensionFromscrathGenerator?

Reply:
Sure, my concern is this generator is not only for the API Plugin